src/main/java/edu/kis/vh/nursery/HanoiRhymer.java

```diff
@@ -1,17 +1,37 @@
 package edu.kis.vh.nursery;
 
-public class HanoiRhymer extends defaultCountingOutRhymer {
+import edu.kis.vh.nursery.stackImpl.IntStack;
 
-int totalRejected = 0;
+public class HanoiRhymer extends DefaultCountingOutRhymer {
 
-	public int reportRejected() {
-		return totalRejected;
-	}
+    private int totalRejected = 0;
 
-	public void countIn(int in) {
-	if (!callCheck() && in > peekaboo())
-			totalRejected++;
-			else
-				super.countIn(in);
-	}
+    public HanoiRhymer() {
+        super();
+    }
+
+    public HanoiRhymer(IntStack intStack) {
+        super(intStack);
+    }
+
+    public int reportRejected() {
+        return getTotalRejected();
+    }
+
+    @Override
+    public void countIn(int in) {
+        if (!callCheck() && in > peekaboo())
+            setTotalRejected(getTotalRejected() + 1);
+        else
+            super.countIn(in);
+    }
+
+    public int getTotalRejected() {

```

Comment:
Shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internally usually do not require encapsulation.